crates/release_plz_core/src/git/backend.rs

```diff
@@ -105,13 +106,26 @@ pub struct GitPr {
     pub head: Commit,
     pub title: String,
     pub body: Option<String>,
+    pub labels: Vec<Label>,
 }
 
 /// Pull request.
 impl GitPr {
     pub fn branch(&self) -> &str {
         self.head.ref_field.as_str()
     }
+
+    pub fn label_names(&self) -> Vec<&str> {
+        self.labels.iter().map(|l| l.name.as_str()).collect()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Label {
+    pub name: String,
+    /// ID of the label.
+    /// Used by Gitea and GitHub. Not present in GitLab responses.
+    id: Option<u64>,
```

Comment:
it would be cool if at compile time with the type system we could have u64 for GitHub and Gitea and Option<u64> for GitLab. But now I don't know how. So if you have a good idea we could do it in another pr :)

Reply:
We could implement something similar to what was done with GitPr and GitLabMr by introducing GitLabel and GitLabLabel. 
Perhaps GitPr could be extended to handle situations where GitLabel would be used because we are dealing with label data returned from a request for PR info. 
Gitea & Github - returns An Object. with  `id`  as one of its keys.
Gitlab - returns only a vector of strings.
So I think extending GitPr and GitLabMr would be the start of the solution. What do you think

Reply:
If you know how to remove the with_context error handling, I would say go for it! But maybe let's do it in another pr, and let's first merge this one 👍